WooCommerce/Classes/ViewRelated/Orders/Order Creation/ProductsSection/ProductRow.swift

```diff
@@ -111,11 +114,14 @@ private enum Localization {
 
 struct ProductRow_Previews: PreviewProvider {
     static var previews: some View {
-        ProductRow(canChangeQuantity: true)
+        let viewModel = ProductRowViewModel(product: ProductRowViewModel.sampleProduct, canChangeQuantity: true)
```

Comment:
I'm a bit concerned about `ProductRowViewModel.sampleProduct`

Previews(as they are views) should not need from a Yosemite/Networking entity in order to work, as it breaks the "view - model" separation of concerns. This makes views highly unreusable in other contexts. Also forces you to expose fake objects that can be accessed by anyone without any control.

You can overcome this by exposing an initializer in your view model with just the information it needs, then it can have a convenience one that receives the real Yosemite/Networking Entity.

Something like

 https://github.com/woocommerce/woocommerce-ios/blob/a39ae7da48c9707b091f0b6a856bc1d0a300e87b/WooCommerce/Classes/ViewRelated/Orders/Simple%20Payments/Summary/SimplePaymentsSummaryViewModel.swift#L81-L135

Reply:
Makes sense! Thanks for the explanation and example. ⭐